modules/reactor/include/meshgenerators/PolygonMeshGeneratorBase.h

```diff
@@ -65,18 +65,49 @@ class PolygonMeshGeneratorBase : public MeshGenerator
     HEXAGON_NUM_SIDES = 6
   };
 
+  /// A struct type used to contain multiple blocks's boundary layer related parameters
+  struct multiBdryLayerParams
+  {
+    std::vector<Real> widths;
+    std::vector<Real> fractions;
+    std::vector<unsigned int> intervals;
+    std::vector<Real> biases;
+  };
+  /// A struct type used to contain a single block's boundary layer related parameters
+  struct singleBdryLayerParams
+  {
+    Real width;
+    unsigned int intervals;
+    Real bias;
+  };
+
 protected:
   /**
    * Creates a mesh of a slice that corresponds to a single side of the polygon to be generated.
    * @param mesh input mesh to build the slice mesh onto
    * @param ring_radii radii of the ring regions
-   * @param rings numbers of radial intervals of the ring regions
+   * @param ring_layers numbers of radial intervals of the ring regions
+   * @param ring_radial_biases values used for radial meshing biasing in ring regions
+   * @param ring_inner_boundary_layer_params widths, radial fractions, radial sectors, and growth
+   * factors of the inner boundary layer of the ring regions
+   * @param ring_outer_boundary_layer_params widths, radial fractions, radial sectors, and growth
+   * factors of the outer boundary layer of the ring regions
    * @param ducts_center_dist distance parameters of the duct regions
    * @param ducts_layers numbers of radial intervals of the duct regions
+   * @param duct_radial_biases values used for radial meshing biasing in duct regions
+   * @param duct_inner_boundary_layer_params widths, radial fractions, radial sectors, and growth
+   * factors of the inner boundary layer of the duct regions
+   * @param duct_outer_boundary_layer_params widths, radial fractions, radial sectors, and growth
+   * factors of the outer boundary layer of the duct regions
    * @param has_rings whether the slice contains ring regions or not
    * @param has_ducts whether the slice contains duct regions or not
    * @param num_sectors_per_side number of azimuthal intervals
    * @param background_intervals number of radial intervals of the background region
+   * @param background_radial_bias value used for radial meshing biasing in background region
```

Comment:
yeah so for background the widths need not be stored explicitly. If you do port rings and ducts to using the struct, let s port background too please!

Reply:
I think it is better that regular region does not use struct as neither width nor fraction can be defined on regular background region